spring-security-web-app/README.md

```diff
@@ -0,0 +1,225 @@
+   - Access token that is validated using Spring security.
+
+### Scenario
+
+This sample shows how to build a Java web app that uses OpenId Connect to sign-in/ sign-out an user and to use spring security to validate access token obtained from the Authorization headers. For more information about how the protocols work in this scenario and other scenarios, see [Authentication Scenarios for Azure AD](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-authentication-scenarios).
+
+### Token Validation
+
+A token represents the outcome of an authentication operation with some artifact that can be unambiguously tied to the Identity Provider that performed the authentication, without relying on any special network infrastructure.
+
+With Azure Active Directory taking the full responsibility of verifying user's raw credentials, the token receiver's responsibility shifts from verifying raw credentials to verifying that their caller did indeed go through your identity provider of choice and successfully authenticated. The identity provider represents successful authentication operations by issuing a token, hence the job now becomes to validate that token.
+
+### What to validate
+
+While you should always validate tokens issued to the resources (audience) that you are developing, your application will also obtain access tokens for other resources from AAD. AAD will provide an access token in whatever token format that is appropriate to that resource.
+This access token itself should be treated like an opaque blob by your application, as your app isn’t the access token’s intended audience and thus your app should not bother itself with looking into the contents of this access token.
+Your app should just pass it in the call to the resource. It's called resource's responsibility to validate this access token.
+
+## How to run this sample
+
+To run this sample, you'll need:
+
+- Working installation of Java and Maven
+- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, see [How to get an Azure AD tenant](https://azure.microsoft.com/en-us/documentation/articles/active-directory-howto-tenant/)
+- One or more user accounts in your Azure AD tenant.
+
+### Step 1: Download Java (8 and above) for your platform
+
+To successfully use this sample, you need a working installation of [Java](https://openjdk.java.net/install/) and [Maven](https://maven.apache.org/).
+
+### Step 2:  Clone or download this repository
+
+From your shell or command line:
+
+- `git clone https://github.com/Azure-Samples/ms-identity-java-webapp.git`
+
+Go to `spring-security-web-app` folder
+
+- cd `spring-security-web-app`
+
+### Step 3:  Register the sample with your Azure Active Directory tenant
+
+To register the project, you can follow the steps in the paragraphs below:
+
+#### First step: choose the Azure AD tenant where you want to create your applications
```

Comment:
This header is repeated on line 71. We can delete this line.

Reply:
yeah sure